src/macosx/osxgl.m

```diff
@@ -1090,6 +1090,26 @@ static CVReturn display_link_callback(CVDisplayLinkRef display_link,
    return kCVReturnSuccess;
 }
 
+/* osx_tell_dock:
+ *  Tell the dock about us; promote us from a console app to a graphical app
+ *  with dock icon and menu
+ */
+static void osx_tell_dock(void)
```

Comment:
Would you mind changing the implementation of the system.m's copy of this function to what you have here, renaming it to `_al_osx_tell_dock` and then calling it from osxgl.m? Trying to avoid copy-pasting here.

The forward declaration for that would go here: `include/allegro5/platform/aintosx.h`